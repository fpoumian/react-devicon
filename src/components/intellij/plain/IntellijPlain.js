
import React from 'react'
import PropTypes from 'prop-types'
import SVGInline from 'react-svg-inline'
import iconSVG from './IntellijPlain.svg'
import { normalizeUnit } from '../../utils.js'

/** IntellijPlain */
function IntellijPlain({width, height}) {
  return (
     <SVGInline
      classSuffix="-devicon"
      className="IntellijPlain"
      svg={iconSVG}
      width={normalizeUnit(width)}
      height={normalizeUnit(height)}
    />
  )
}

IntellijPlain.propTypes = {
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
}

export default IntellijPlain
