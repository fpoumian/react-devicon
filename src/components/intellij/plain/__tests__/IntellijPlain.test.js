import React from 'react'
import renderer from 'react-test-renderer'
import IntellijPlain from '../index.js'

it('renders correctly', () => {
  const tree = renderer.create(<IntellijPlain />).toJSON()
  expect(tree).toMatchSnapshot()
})
