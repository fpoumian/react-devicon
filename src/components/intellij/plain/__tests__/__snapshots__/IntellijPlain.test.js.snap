// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  className="SVGInline IntellijPlain"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg class=\\"SVGInline-devicon IntellijPlain-devicon\\" id=\\"intelij\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 128 128\\"><defs><style>.cls-1{fill:#136ba2}</style></defs><title>Artboard 15</title><g id=\\"plain\\"><path class=\\"cls-1\\" d=\\"M4.87 64.74a17.28 17.28 0 1 0 17.28-17.29A17.31 17.31 0 0 0 4.87 64.74zM108 4c-9.5 0-15.16 7.65-15.16 20.46v78.7c0 12.81 5.65 20.46 15.14 20.46s15.16-7.65 15.16-20.46V24.51C123.13 11.7 117.47 4 108 4z\\"/><path class=\\"cls-1\\" d=\\"M81.45 5.71H19.37a15.09 15.09 0 0 0 0 30.16V36h31.77V64.74a29 29 0 0 1-29 29h-.78l-.08.1c-.43 0-.86-.07-1.3-.07A15.11 15.11 0 1 0 20 124c.45 0 .91 0 1.35-.07v.05h.79a59.33 59.33 0 0 0 59.31-59.24v-.55-9.7-.11V5.71z\\"/></g></svg>",
    }
  }
/>
`;
